exporter/splunkexporter/metricdata_to_splunk.go

```diff
@@ -0,0 +1,46 @@
+package splunkexporter
+
+import (
+	"fmt"
+	"go.opentelemetry.io/collector/consumer/consumerdata"
+	"go.uber.org/zap"
+)
+
+type splunkMetric struct {
+	Time       int64             `json:"time"`                 // epoch time
+	Host       string            `json:"host"`                 // hostname
+	Source     string            `json:"source,omitempty"`     // optional description of the source of the event; typically the app's name
+	SourceType string            `json:"sourcetype,omitempty"` // optional name of a Splunk parsing configuration; this is usually inferred by Splunk
+	Index      string            `json:"index,omitempty"`      // optional name of the Splunk index to store the event in; not required if the token has a default index set in Splunk
+	Event      string            `json:"event"`                // type of event: this is a metric.
+	Fields     map[string]interface{} `json:"fields"`               // metric data
+}
+
+func metricDataToSplunk(logger *zap.Logger, data consumerdata.MetricsData, config Config) ([]*splunkMetric, int, error) {
+	// this is wrong
+	host := data.Node.ServiceInfo.Name
+	if host == "" {
+		host = "unknown"
+	}
+	splunkMetrics := make([]*splunkMetric, 0)
+	for _, metric := range data.Metrics {
+		for _, timeSeries := range metric.Timeseries {
+			for _, tsPoint := range timeSeries.Points {
+				sm := &splunkMetric{
+					Time: tsPoint.Timestamp.GetSeconds(),
+					Host: host,
+					Source: config.Source,
+					SourceType: config.SourceType,
+					Index: config.Index,
+					Event: "metric",
```

Comment:
It would be nice to define "metric" as a const somewhere near the `splunkMetric` type and give it a descriptive name.

However, I cannot find in [Splunk docs](https://docs.splunk.com/Documentation/Splunk/8.0.3/Data/Sendmetricstoametricsindex) that this is expected for metric data. Can you confirm that this is how metric data is expected to be sent? If you have a link that describes this it would be great.

Reply:
Yes, I found this on this page here: https://docs.splunk.com/Documentation/Splunk/8.0.3/Metrics/GetMetricsInOther
See paragraph "Send data to a metrics index over HTTP"
`Metrics data, which requires an "event" field set to "metric".`

Reply:
I have moved `metric` to a constant.